src/services/auth.js

```diff
@@ -48,6 +49,20 @@ passport.use(new HTTPSignatureStrategy(
       })
   }))
 
+passport.use(new ClientCertStrategy((userName, info, done) => {
+  if (!config.auth.client_certificates_enabled) {
+    return done(new UnauthorizedError('Unsupported authentication method'))
+  }
+
+  Account.findByName(userName)
+    .then(function (userObj) {
+      if (!userObj || userObj.is_disabled || userObj.fingerprint !== info.fingerprint) {
```

Comment:
> I think we could only use one authentication method at a time.

I don't understand. The username is the same regardless of authentication method. If it's the fingerprint, then what happens to a user who doesn't use client certs?